pkg/util/kubernetes/apiserver/controllers/metadata_controller.go

```diff
@@ -78,7 +78,13 @@ func (m *metadataController) run(stopCh <-chan struct{}) {
 		return
 	}
 
-	filter := workloadmeta.NewFilterBuilder().AddKind(workloadmeta.KindKubernetesNode).Build()
+	filter := workloadmeta.NewFilterBuilder().AddKindWithEntityFilter(
```

Comment:
Discussed with @adel121 to try to find a better way to write this kind of filter in a future PR.